src/overlays/actors/ovl_Magic_Dark/z_magic_dark.c

```diff
@@ -66,37 +69,218 @@ void MagicDark_Destroy(Actor* thisx, GlobalContext* globalCtx) {
     }
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Magic_Dark/func_80B874E4.s")
+void MagicDark_DiamondUpdate(Actor* thisx, GlobalContext* globalCtx) {
+    MagicDark* this = THIS;
+    u8 phi_a0;
+    Player* player = PLAYER;
+    u8 temp;
+    s16 nayrusLoveTimer = gSaveContext.nayrusLoveTimer;
+    s32 msgMode = globalCtx->msgCtx.msgMode;
+    u8 temp2;
+
+    if (1) {}
+
+    if ((msgMode == 0xD) || (msgMode == 0x11)) {
+        Actor_Kill(thisx);
+        return;
+    }
+
+    if (nayrusLoveTimer >= 1200) {
+        player->invincibilityTimer = 0;
+        gSaveContext.nayrusLoveTimer = 0;
+        Actor_Kill(thisx);
+        return;
+    }
+
+    player->invincibilityTimer = -100;
+    thisx->scale.x = thisx->scale.z = this->scale;
+
+    if (this->timer < 20) {
+        thisx->scale.x = thisx->scale.z = (1.6f - (this->timer * 0.03f)) * this->scale;
+        thisx->scale.y = ((this->timer * 0.01f) + 0.8f) * this->scale;
+    } else {
+        thisx->scale.x = thisx->scale.z = this->scale;
+        thisx->scale.y = this->scale;
+    }
+
+    thisx->scale.x *= 1.3f;
+    thisx->scale.z *= 1.3f;
+
+    phi_a0 = (this->timer < 20) ? (this->timer * 12) : 255;
+
+    if (nayrusLoveTimer >= 1180) {
+        this->primAlpha = 15595 - (nayrusLoveTimer * 13);
+        if (nayrusLoveTimer & 1) {
+            temp2 = this->primAlpha;
+            this->primAlpha = (temp2 & 0xFF) >> 1;
+        }
+    } else {
+        if (nayrusLoveTimer >= 1100) {
+            temp = (nayrusLoveTimer << 7);
+            temp += 127;
+            this->primAlpha = temp;
```

Comment:
Also, maybe this works?
```suggestion
            this->primAlpha = (u8)(nayrusLoveTimer << 7) + 127;
```